path_oidc.go

```diff
@@ -379,3 +499,15 @@ func validRedirect(uri string, allowed []string) bool {
 
 	return false
 }
+
+// parseMount attempts to extract the mount path from a redirect URI.
+func parseMount(redirectURI string) string {
```

Comment:
Bonus - would be nice to see an example string here.

Reply:
For this one, I wrote a test on a branch like this:
```
func TestParseMount(t *testing.T) {
	result := parseMount("https://example.com/v1/auth/oidc/foo")
	fmt.Println(result)
}
```
When I run that, it prints out "oidc". It looks like the mount path _can_ include any ascii characters, and "/" is one of them. (https://www.vaultproject.io/api/system/mounts.html#inlinecode-path-14.)

If I change the input to "https://example.com/v1/auth/oidc", I still get "oidc". So it definitely works in the most common use but I think we need to change the algorithm in here a little.

Reply:
This is addressed in https://github.com/hashicorp/vault-plugin-auth-jwt/pull/99 if we accept it.